### PR TITLE
284 related articles UI block

### LINKED ISSUE #284: [Blog/News] Implement "Related Articles" blog based on keywords
This must be implemented on two places:

* Autoblock to add the "related-articles" block in https://github.com/hlxsites/danaher-ls-aem/blob/main/templates/blog/blog.js at the end before the social media links
* Implement the "related-articles" block to display the cards based on the index.

### PR DESCRIPTION
**Pull request by rgravitvl**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #284 

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/
- After: https://284-related-articles-ui-block--danaher-ls-aem--hlxsites.hlx.page/us/en/blog/industrializing-biology-biopharma-4
- After: https://284-related-articles-ui-block--danaher-ls-aem--hlxsites.hlx.page/us/en/blog/how-organoids-can-redefine-pre-clinical-research
- After: https://284-related-articles-ui-block--danaher-ls-aem--hlxsites.hlx.page/us/en/blog/lab-tools-enhance-mrna-development-manufacturing-efficiency
- After: https://284-related-articles-ui-block--danaher-ls-aem--hlxsites.hlx.page/us/en/blog/the-ai-powered-pixel-classifier

### CODE REVIEW

File: blocks/related-articles/related-articles.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    class:
+        'w-full flex flex-col col-span-1 relative mx-auto justify-center transform transition duration-500 border hover:scale-105 shadow-lg rounded-lg overflow-hidden bg-white max-w-xl',
+  }, cardWrapper);
+};
+
+export default async function decorate(block) {
+  const articleType = getMetadata('template').toLowerCase();
+  const articleKeyword = getMetadata('keywords')?.toLowerCase();
+  const url = new URL(getMetadata('og:url'));
+  let articles = await ffetch('/us/en/query-index.json')
```
Comment (mhaack) [inline blocks/related-articles/related-articles.js:48]: This should now be `article-index` :-)

File: scripts/scripts.js [hunk 2]
```diff
@@ -24,6 +29,31 @@ const TEMPLATE_LIST = {
...
 };
 
+/**
+ * Get the Image URL from Scene7 and Optimize the picture
+ * @param {string} imageUrl
+ * @param {string} imageAlt
+ * @param {boolean} eager
+ * @returns Optimized image
+ */
+export function imageHelper(imageUrl, imageAlt, eager = false) {
```
Comment (mhaack) [inline scripts/scripts.js:39]: This looks similar to https://github.com/hlxsites/danaher-ls-aem/blob/main/blocks/card-list/card-list.js#L11 please update card-list as well to use the method moved to scripts.js

File: blocks/related-articles/related-articles.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import ffetch from '../../scripts/ffetch.js';
+import { getMetadata } from '../../scripts/lib-franklin.js';
+import {
+  formatDateUTCSeconds, makePublicUrl, imageHelper,
+} from '../../scripts/scripts.js';
+import {
+  ul, li, a, p, time, span, h2,
+} from '../../scripts/dom-builder.js';
+
+const createCard = (article, firstCard = false) => {
```
Comment (mhaack) [inline blocks/related-articles/related-articles.js:10]: Is this a copy of https://github.com/hlxsites/danaher-ls-aem/blob/main/blocks/card-list/card-list.js#L29C9-L29C9? we should export the function on card-list and import it here.

**Review by mhaack — CHANGES_REQUESTED** [review]

File: blocks/related-articles/related-articles.js (hunk 3 above)
Comment (rgravitvl) [reply blocks/related-articles/related-articles.js:10]: updated the style a bit and removed read article text, any way I will confirm the changes with @davenichols-DHLS

Comment (mhaack) [reply blocks/related-articles/related-articles.js:10]: Maybe we can have one createCard method which supports both styles with some optional parameters.

File: blocks/related-articles/related-articles.js (hunk 1 above)
Comment (rgravitvl) [reply blocks/related-articles/related-articles.js]: Done

File: scripts/scripts.js (hunk 2 above)
Comment (rgravitvl) [reply scripts/scripts.js:39]: Done

File: blocks/related-articles/related-articles.js (hunk 3 above)
Comment (rgravitvl) [reply blocks/related-articles/related-articles.js]: @mhaack As discussed with @davenichols-DHLS , we are good with Card from hub page. So I just exported the createCard from card-list.

**Review by mhaack — APPROVED** [review]

**Pull request merged**